### PR TITLE
Add get_user_id

### PR DESCRIPTION
**Pull request by noeliarico**
As most of the functions work with the user id, this function retrieves the user ids for given a vector of user names

### CODE REVIEW

File: R/get_user_id.R [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#' Get user id
+#' 
+#' This function get the user ID given a user name.
+#'
+#' @param username string containing the screen name of the user
```
Comment (chainsawriot) [inline R/get_user_id.R:5]: usernames

File: R/get_user_id.R [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    "usernames" = paste(usernames, collapse = ",")
+  )
+  
+  dat <- make_query(url = url,
+                    params = params,
+                    bearer_token = bearer_token, verbose = TRUE)
+
+  dat <- dat[[1]] # results are returned in a list
+  
+  if(all) return(dat)
```
Comment (chainsawriot) [inline R/get_user_id.R:34]: May I suggest something like this:

```r
if (all) {
   return(dat) # it got return, so nothing else matters
}
ids <- dat$id
names(ids) <- usernames
return(ids)
```

File: R/get_user_id.R [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+                    params = params,
+                    bearer_token = bearer_token, verbose = TRUE)
+
+  dat <- dat[[1]] # results are returned in a list
+  
+  if (all) {
+    return(dat)
+  }
+  ids <- dat$id
+  names(ids) <- usernames
```
Comment (chainsawriot) [inline R/get_user_id.R:38]: This will fail if the returned `dat` does not have the same length as `usernames`.

**Comment by chainsawriot** [comment]
@noeliarico Thanks for the PR. I will also need to add some test cases. In the meantime, is it [your ORCID](https://orcid.org/0000-0002-6169-4523)? We need to add you to the DESCRIPTION

**Comment by cjbarrie** [comment]
@noeliarico thanks for the PR! I was giving a tutorial today when I realised that we could make getting IDs easier to do. Thanks a lot for your contribution

**Comment by chainsawriot** [comment]
@noeliarico ok, I have added some test cases. But I have not added your name to DESCRIPTION yet. Please confirm your ORCID. After that then I will merge.

**Comment by noeliarico** [comment]
I'm very glad that you find the function useful 😃 

And yes, that is my ORCID. Thanks a lot for adding me!

**Pull request merged**